### PR TITLE
AWS CMWI GA

### LINKED ISSUE #3967: Customer Managed Identity Support on Astro Hosted (AWS) is GA
### Issue Type

Inaccurate, misleading, or out-of-date information

### Links to Affected Docs

https://www.astronomer.io/docs/astro/release-notes#customer-managed-workload-identity-for-aws

### The Issue or Context

The feature is now GA. We can drop the public preview banner, and make this the recommended option.

Additionally, lets add a configuration example that allows for specificity of the kubernetes service accounts for organizations that do not allow wildcard use in Trust Policies. Example config would be:

```
{
    "Condition": {
        "StringLike": {
            "<clusterOIDCIssuerUrl>:aud": "sts.amazonaws.com",
            "<clusterOIDCIssuerUrl>:sub": "system:serviceaccount:<DeploymentNamespace>:<DeploymentNamespace>-kpo"
            "<clusterOIDCIssuerUrl>:sub": "system:serviceaccount:<DeploymentNamespace>:<DeploymentNamespace>-scheduler-serviceaccount"
            "<clusterOIDCIssuerUrl>:sub": "system:serviceaccount:<DeploymentNamespace>:<DeploymentNamespace>-triggerer-serviceaccount"
            "<clusterOIDCIssuerUrl>:sub": "system:serviceaccount:<DeploymentNamespace>:<DeploymentNamespace>-webserver-serviceaccount"
            "<clusterOIDCIssuerUrl>:sub": "system:serviceaccount:<DeploymentNamespace>:<DeploymentNamespace>-worker-serviceaccount"
        }
    }
}
```

### Fix or Proposed Change

_No response_

### Additional Notes

_No response_

### Required Reviewers

chomipi88

### PR DESCRIPTION
**Pull request by lzdanski**
Resolves #3967

### CODE REVIEW

**Review by chomipi88 — COMMENTED** [review]
LGTM. Can we add the additional configuration described in the issue?

File: astro/authorize-deployments-to-your-cloud.md [hunk 1]
```diff
@@ -45,7 +45,101 @@ values={[
...
+</details>
+
+5. Enter your IAM role ARN when prompted, then copy and run the provided CLI command.
+6. Click **Update Deployment**. The IAM role is now selectable as a workload identity for the Deployment.
+
+Repeat these steps for each Astro Deployment that needs to access your AWS resources.
+
+#### Step 2: Create an Airflow connection
+
+Now that your Deployment is authorized, you can connect it to your cloud using an Airflow connection. Create an **Amazon Web Services** connection in either the [Astro UI](create-and-link-connections.md) or the Airflow UI for your Deployment, and specify the following fields:
```
Comment (merobi-hub) [inline astro/authorize-deployments-to-your-cloud.md:123]:
```suggestion
Now that your Deployment is authorized, you can connect it to your cloud using an Airflow connection. Create an **Amazon Web Services** connection in either the [Astro UI](create-and-link-connections.md) or the Airflow UI for your Deployment and specify the following fields:
```

File: astro/authorize-deployments-to-your-cloud.md [hunk 2]
```diff
@@ -45,7 +45,101 @@ values={[
...
+
+- **Connection Id**: Enter a name for the connection.
+
+If you don't see **Amazon Web Services** as a connection type in the Airflow UI, ensure you have installed its provider package in your Astro project's `requirements.txt` file. See **Use Provider** in the [Astronomer Registry](https://registry.astronomer.io/providers/Amazon/versions/latest) for the latest package.
+
+:::tip
+
+If you use a mix of strategies for managing connections, if you define the same connection in multiple ways, Airflow uses the following order of precedence:
```
Comment (merobi-hub) [inline astro/authorize-deployments-to-your-cloud.md:131]:
```suggestion
If you use a mix of strategies for managing connections and define the same connection in multiple ways, Airflow uses the following order of precedence:
```

**Review by merobi-hub — APPROVED** [review]

File: astro/authorize-deployments-to-your-cloud.md [hunk 3]
```diff
@@ -45,7 +45,101 @@ values={[
...
+  - [openSSL](https://www.openssl.org/source/)
+
+#### Step 1: Authorize the Deployment to your IAM role
+
+To authorize your Deployment, create an IAM role to assign as your Deployment's workload identity:
+
+1. [Create an IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-service.html) to delegate permissions to in an AWS service. Grant the role any permission that the Deployment will need in your AWS account. Copy the IAM role ARN to use later in this setup.
+2. In the Astro UI, select your Deployment and then click **Details**. In the **Advanced** section, click **Edit**.
+3. In the **Workload Identity** menu, select **Customer Managed Identity**.
+4. (Optional - Dedicated clusters only) Customize your managed identity to specify Kubernetes service accounts or use a wildcard so you can share or re-use managed identity information.
```
Comment (chomipi88) [inline astro/authorize-deployments-to-your-cloud.md:77]: Let's move the optional configs to a separate step. Focus on the easy path.

File: astro/authorize-deployments-to-your-cloud.md [hunk 4]
```diff
@@ -45,7 +45,101 @@ values={[
...
+
+<details>
+  <summary><strong>Specify Kubernetes service accounts</strong></summary>
```
Comment (chomipi88) [inline astro/authorize-deployments-to-your-cloud.md:80]: The `Specify Kubernetes service accounts` optional config can be used for Standard or Dedicated

File: astro/authorize-deployments-to-your-cloud.md [hunk 5]
```diff
@@ -45,7 +45,101 @@ values={[
...
+              "<clusterOIDCIssuerUrl>:sub": "system:serviceaccount:<DeploymentNamespace>:<DeploymentNamespace>-webserver-serviceaccount"
+              "<clusterOIDCIssuerUrl>:sub": "system:serviceaccount:<DeploymentNamespace>:<DeploymentNamespace>-worker-serviceaccount"
+          }
+      }
+  }
+  ```
+</details>
+
+<details>
+  <summary><strong>Share or re-use a managed identity using a wildcard</strong></summary>
```
Comment (chomipi88) [inline astro/authorize-deployments-to-your-cloud.md:100]: The `Share or re-use a managed identity using a wildcard` optional config should **only** be used on Dedicated for security purposes

File: astro/authorize-deployments-to-your-cloud.md [hunk 6]
```diff
@@ -45,7 +45,101 @@ values={[
...
+To authorize your Deployment, create an IAM role to assign as your Deployment's workload identity:
+
+1. [Create an IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-service.html) to delegate permissions to in an AWS service. Grant the role any permission that the Deployment will need in your AWS account. Copy the IAM role ARN to use later in this setup.
+2. In the Astro UI, select your Deployment and then click **Details**. In the **Advanced** section, click **Edit**.
+3. In the **Workload Identity** menu, select **Customer Managed Identity**.
+4. (Optional - Dedicated clusters only) Customize your managed identity to specify Kubernetes service accounts or use a wildcard so you can share or re-use managed identity information.
+
+<details>
+  <summary><strong>Specify Kubernetes service accounts</strong></summary>
+If you want to share or re-use the same customer managed identity on static or ephemeral Deployments for dedicated clusters, and your organization does not allow you to use a wildcard, change the `<DeploymentNamespace>` value in `Condition` to specify the Kubernetes service accounts. The following shows an example:
```
Comment (chomipi88) [inline astro/authorize-deployments-to-your-cloud.md:81]:
```suggestion
If your organization does not allow you to use a wildcards in your IAM Trust Policies, change the `<DeploymentNamespace>` value in `Condition` to specify the Kubernetes service accounts. The following shows an example:
```

File: astro/authorize-deployments-to-your-cloud.md [hunk 7]
```diff
@@ -138,30 +76,114 @@ To authorize your Deployment, create an IAM role to assign as your Deployment's
 3. In the **Workload Identity** menu, select **Customer Managed Identity**.
 4. Enter your IAM role ARN when prompted, then copy and run the provided CLI command.
```
Comment (chomipi88) [inline astro/authorize-deployments-to-your-cloud.md:77]: Let's add click `Save Configuration` to have it selectable.

File: astro/authorize-deployments-to-your-cloud.md [hunk 8]
```diff
@@ -138,30 +76,114 @@ To authorize your Deployment, create an IAM role to assign as your Deployment's
...
 5. Click **Update Deployment**. The IAM role is now selectable as a workload identity for the Deployment.
```
Comment (chomipi88) [inline astro/authorize-deployments-to-your-cloud.md:78]: By clicking Update Deployment, it applies the selected role to the Deployment.

**Review by chomipi88 — APPROVED** [review]
couple small updates, otherwise LGTM

File: astro/authorize-deployments-to-your-cloud.md (hunk 8 above)
Comment (lzdanski) [inline astro/authorize-deployments-to-your-cloud.md:78]:
```suggestion
5. Click **Update Deployment** to apply the selected IAM role to the Deployment.
```

File: astro/authorize-deployments-to-your-cloud.md (hunk 7 above)
Comment (lzdanski) [inline astro/authorize-deployments-to-your-cloud.md:77]:
```suggestion
4. Enter your IAM role ARN when prompted, then copy and run the provided CLI command. Click **Save Configuration** to save the IAM role as a selectable configuration.
```

**Pull request merged**